src/window_base.cpp

```diff
@@ -320,15 +319,18 @@ void Window_Base::DrawGauge(const Game_Battler& actor, int cx, int cy) const {
 	Rect gauge_center(16, gauge_y, 16, 16);
 	Rect gauge_right(32, gauge_y, 16, 16);
 
-	// Full or not full bar
-	Rect gauge_bar(full ? 64 : 48, gauge_y, 16, 16);
-
 	Rect dst_rect(cx + 16, cy, 25, 16);
-	Rect bar_rect(cx + 16, cy, gauge_w, 16);
 
 	contents->Blit(cx + 0, cy, *system2, gauge_left, 255);
 	contents->Blit(cx + 16 + 25, cy, *system2, gauge_right, 255);
-
 	contents->StretchBlit(dst_rect, *system2, gauge_center, 255);
-	contents->StretchBlit(bar_rect, *system2, gauge_bar, 255);
+
+	const auto atb = actor.GetAtbGauge();
+	if (atb > 0) {
+		// Full or not full bar
+		int gauge_w = 25 * atb / actor.GetMaxAtbGauge();
```

Comment:
This still div/0 in StretchBlit: gauge_w can become 0 here.
Then zoom_x is inf: ``src_rect.width  / dst_rect.width``

Reply:
Are you sure? This logic is skipped if atb is 0 and `GetMaxAtbGauge()` is a constexpr.

Reply:
25 * 280 / 300000 is still 0

Reply:
🤦